include/preprocess/max_inscribed_ellipsoid.hpp

```diff
@@ -156,9 +167,9 @@ std::pair<std::pair<MT, VT>, bool> max_inscribed_ellipsoid(MT A, VT b, VT const&
         h_z = h + z;
 
         for (int j = 0; j < n; ++j) {
-            T.col(j) = G.colPivHouseholderQr().solve(YA.col(j).cwiseProduct(h_z));
+            T.col(j) = G.colPivHouseholderQr().solve( VT(YA.col(j).cwiseProduct(h_z)) );
         }
-        ATP.noalias() = (y2h.asDiagonal()*T - YA).transpose();
+        ATP.noalias() = MT(y2h.asDiagonal()*T - YA).transpose();
```

Comment:
Since y2h is a diagonal matrix, could this operation be further optimized?

Reply:
here also, as `y2h.asDiagonal()` tells Eigen to treat it as a diagonal matrix and optimize

Reply:
is it needed to declare the result as `MT`? similar for `VT` above

Reply:
Yup, this is needed, because `YA` is of type `Custom_MT` and `ATP` is of type `MT` so explicit casting is required.